sylt-compiler/src/typechecker.rs

```diff
@@ -14,30 +14,55 @@ use std::collections::{BTreeMap, BTreeSet};
 
 type TypeResult<T> = Result<T, Vec<Error>>;
 
+trait Help {
+    fn help(self, typechecker: &TypeChecker, span: Span, message: String) -> Self;
+}
+
+impl<T> Help for TypeResult<T> {
+    fn help(mut self, typechecker: &TypeChecker, span: Span, message: String) -> Self {
+        match &mut self {
+            Ok(_) => {}
+            Err(errs) => match &mut errs.last_mut() {
+                Some(Error::TypeError { helpers, .. }) => {
+                    helpers.push(Helper {
+                        file: typechecker.namespace_to_file[&span.file_id].clone(),
```

Comment:
There must be some way of removing this...

Reply:
If we could have a global life-time, we could remove this. (But we'd need to mutate global state which is unsafe and will make Gustav very very sad)